test/gtest/uct/ib/test_ib_md.cc

```diff
@@ -310,4 +310,64 @@ UCS_TEST_P(test_ib_md, smkey_reg_atomic_mt, "IB_REG_MT_THRESH=1k",
     test_smkey_reg_atomic();
 }
 
+UCS_TEST_SKIP_COND_P(test_ib_md, mt_fail,
+                     has_ksm() &&
+                             check_invalidate_support(UCT_MD_MEM_ACCESS_RMA),
+                     "IB_REG_MT_THRESH=128K", "IB_REG_MT_CHUNK=16K")
+{
+    size_t size             = UCS_MBYTE;
+    const size_t align_mask = (8 * UCS_KBYTE) - 1;
+    size_t lower_size       = ((size / 3)) & ~align_mask;
+    size_t upper_size       = (size - (size / 2)) & ~align_mask;
+    void *start, *mid, *last;
+    uct_mem_h memh;
+
+    auto mmap_anon = [](void *ptr, size_t size) {
+        ptr = mmap(ptr, size, PROT_READ | PROT_WRITE,
+                   MAP_PRIVATE | MAP_ANONYMOUS |
+                           ((ptr != nullptr) ? MAP_FIXED : 0),
+                   -1, 0);
+        return ptr != MAP_FAILED ? ptr : nullptr;
+    };
+
+    /* Find an available VMA */
+    start = mmap_anon(nullptr, size);
+    ASSERT_NE(nullptr, start);
+    mid  = reinterpret_cast<char*>(start) + lower_size;
+    last = reinterpret_cast<char*>(start) + size - upper_size;
+    munmap(start, size);
+
+    /* Allocate lower half and upper half */
+    /* coverity[pass_freed_arg] */
+    start = mmap_anon(start, lower_size);
+    last  = mmap_anon(last, upper_size);
+    EXPECT_NE(nullptr, start);
```

Comment:
1. move it right after mmap_anon(start, lower_size)
2. maybe better just munmap between "start+lower_size" until "last"?
 you assume that after we munmap the block in line 338, no other thread will try to allocate in that area

Reply:
fixed both